File: MLT_Project/feat_selection.py
import numpy as np
import scipy.io as data
dict = data.loadmat("C:\Users\BHANU YADAV\Documents\MLT_Project\fmri_words.mat")
X_train = np.array(dict["X_train"])
Y_train = np.array(dict["Y_train"])
X_test = np.array(dict["X_test"])
Y_test = np.array(dict["Y_test"])
word_features_std = np.array(dict["word_features_std"])
X_train
print word_features_std
for i in (1, 300):
    S(i) = word_features_std(Y_train(i))
end
for i in (1, 218):
	for j in (1, 21764):
		B = np.corrcoef(X_train(all, j), S(all, i))
		C(i, j) = B(1, 2)

File: MLT_Project/feat_selection.m
clear all;
load('fmri_words.mat');
count = zeros(1,21764);
z = 10000;
for p = 1:300
    S(p,:) = word_features_std(Y_train(p),:);
end
for i = 1:218
	for j = 1:21764
        %X_train(:,j)
        %S(:,i)
		B = corrcoef(X_train(:,j),S(:,i));
		C(j,i) = B(1,2);
    end
end
%for m = 1:21764
   [~,I] = sort(C,'descend');
%end
for r = 1:218
    for q = 1:z
        count(I(q,r)) = count(I(q,r)) + 1;
    end
end
[~,I_new] = sort(count,'descend');
for k = 1:300
    for l = 1:z
        X_train_new(k,l) = X_train(k,I_new(l));
    end
end